tools/tpm2_createek.c

```diff
@@ -194,8 +196,17 @@ static tool_rc set_ek_template(ESYS_CONTEXT *ectx, TPM2B_PUBLIC *input_public) {
         nonce_nv_index = RSA_EK_NONCE_NV_INDEX;
         break;
     case TPM2_ALG_ECC:
-        template_nv_index = ECC_EK_TEMPLATE_NV_INDEX;
-        nonce_nv_index = ECC_EK_NONCE_NV_INDEX;
+        if (input_public->publicArea.parameters.eccDetail.curveID == TPM2_ECC_NIST_P256) {
+            template_nv_index = ECC_EK_TEMPLATE_NV_INDEX;
+            nonce_nv_index = ECC_EK_NONCE_NV_INDEX;
+        } else if (input_public->publicArea.parameters.eccDetail.curveID == TPM2_ECC_SM2_P256) {
+            template_nv_index = ECC_SM2_EK_TEMPLATE_NV_INDEX;
+            // EK Nonces SHALL NOT be Populated in any NV Index in the High Range.
```

Comment:
is this specific to SM2?

Reply:
I belioeve its all high range NV indices